packages/medusa/src/services/cart.ts

```diff
@@ -315,66 +200,74 @@ class CartService extends TransactionBaseService {
    * Gets a cart by id.
    * @param cartId - the id of the cart to get.
    * @param options - the options to get a cart
+   * @param totalsConfig
    * @return the cart document.
    */
   async retrieve(
     cartId: string,
     options: FindConfig<Cart> = {},
     totalsConfig: TotalsConfig = {}
   ): Promise<Cart> {
+    const { totalsToSelect } = this.transformQueryForTotals_(options)
+
+    if (totalsToSelect.length) {
+      return await this.retrieveLegacy(cartId, options, totalsConfig)
+    }
+
     const manager = this.manager_
     const cartRepo = manager.getCustomRepository(this.cartRepository_)
 
-    const { select, relations, totalsToSelect } =
-      this.transformQueryForTotals_(options)
-
-    const query = buildQuery({ id: cartId }, { ...options, select, relations })
+    const query = buildQuery({ id: cartId }, options)
 
-    if (relations && relations.length > 0) {
-      query.relations = relations
+    if ((options.select || []).length <= 0) {
```

Comment:
suggestion(non-blocking): 
```suggestion
    if ((options.select || []).length === 0) {
```